src/transformers/tokenization_pegasus.py

```diff
@@ -0,0 +1,124 @@
+from typing import Dict, List, Optional
+
+from transformers.tokenization_reformer import ReformerTokenizer
+
+from .tokenization_utils_base import BatchEncoding
+
+
+class PegasusTokenizer(ReformerTokenizer):
+    offset = 103  # to make embedding size a multiple of 128 I think
+    vocab_files_names = {"vocab_file": "spiece.model"}
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # Dont use reserved words added_token_encoder, added_tokens_decoder because they
+        # get checked in from_pretrained
+        assert len(self.added_tokens_decoder) == 0
+        self.encoder: Dict[int, str] = {0: self.pad_token, 1: self.eos_token}
+        self.encoder.update({i: f"unk_{i}" for i in range(2, self.offset + 2)})
+        # self.added_tokens_decoder = self._added_toks
+        self.decoder: Dict[str, int] = {v: k for k, v in self.encoder.items()}
+        assert 104 in self.encoder
+        assert 105 not in self.encoder
+        assert self.pad_token_id == 0, "pad should be 0"
+        assert self.eos_token_id == 1, "eos should be 1"
```

Comment:
it seems incoherent to me to allow these values to be modified in the configuration if they can't be modified in the tokenizer

Reply:
done, asserts moved to testing file, only checks pretrained tokenizers.